yt_dlp/extractor/zdf.py

```diff
@@ -316,12 +357,28 @@ def _extract_entry(self, url, player, content, video_id):
             'timestamp': unified_timestamp(content.get('editorialDate')),
             'thumbnails': thumbnails,
             'chapters': chapters or None,
+            'episode': title,
+            **traverse_obj(content, ('programmeItem', 0, 'http://zdf.de/rels/target', {
+                'series_id': ('http://zdf.de/rels/cmdm/series', 'seriesUuid'),
+                'series': ('http://zdf.de/rels/cmdm/series', 'seriesTitle'),
+                'season': ('http://zdf.de/rels/cmdm/season', 'seasonTitle'),
+                'season_number': ('http://zdf.de/rels/cmdm/season', 'seasonNumber'),
+                'season_id': ('http://zdf.de/rels/cmdm/season', 'seasonUuid'),
+                'episode_number': 'episodeNumber',
+                'episode_id': 'contentId',
+            })),
         })
 
     def _extract_regular(self, url, player, video_id):
-        content = self._call_api(
-            player['content'], video_id, 'content', player['apiToken'], url)
-        return self._extract_entry(player['content'], player, content, video_id)
+        player_content_v2 = player['content']
+        player_content_v3 = update_url_query(player_content_v2, {'profile': 'player-3'})
+
+        content = try_call(
+            lambda: self._call_api(player_content_v3, video_id, 'content', player['apiToken'], url),
+            lambda: self._call_api(player_content_v2, video_id, 'content', player['apiToken'], url),
+        )
+
+        return self._extract_entry(player_content_v2, player, content, video_id)
```

Comment:
I don't like how `try_call` is suppressing all the error output and then we're passing `None` as the value of `content` to `_extract_entry()` if both calls fail. I would do this:

```suggestion
        player_url = player['content']

        for query in ({'profile': 'player-3'}, None):
            try:
                content = self._call_api(
                    update_url_query(player_url, query),
                    video_id, 'content', player['apiToken'], url)
            except ExtractorError as e:
                if query is None:
                    raise
                self.report_warning(f'{e.orig_msg}; retrying with profile v2')

        return self._extract_entry(player_url, player, content, video_id)
```

Reply:
I see, I was not aware of that, good catch

Reply:
your code suggestion did not do the trick, it always calls v2, either we need a break there or we remove the loop and call v2 in the catch block (I did this now)